authority/authorize.go

```diff
@@ -72,11 +73,23 @@ func (a *Authority) authorizeToken(ott string) (provisioner.Interface, error) {
 		reuseKey = claims.Nonce
 	}
 	if reuseKey != "" {
-		if _, ok := a.ottMap.LoadOrStore(reuseKey, &idUsed{
-			UsedAt:  time.Now().Unix(),
-			Subject: claims.Subject,
-		}); ok {
-			return nil, &apiError{errors.Errorf("authorizeToken: token already used"), http.StatusUnauthorized, errContext}
+		switch a.db.(type) {
+		case *db.NoopDB:
+			if _, ok := a.ottMap.LoadOrStore(reuseKey, &idUsed{
+				UsedAt:  time.Now().Unix(),
+				Subject: claims.Subject,
+			}); ok {
+				return nil, &apiError{errors.Errorf("authorizeToken: token already used"), http.StatusUnauthorized, errContext}
+			}
```

Comment:
To avoid this type switch we could implement the ott map in the db.NoopDB, yes it will look a little weird, but it will be nicer here.